dynawo/sources/Models/Modelica/Dynawo/Electrical/HVDC/HvdcPTanPhi/HvdcPTanPhiDangling.mo

```diff
@@ -0,0 +1,91 @@
+    input Types.ActivePowerPu P1RefPu (start = s10Pu.re) "Active power regulation set point in p.u (base SnRef) at terminal 1";
+
+    parameter Types.ReactivePowerPu Q1MinPu  "Minimum reactive power in p.u (base SnRef) at terminal 1";
+    parameter Types.ReactivePowerPu Q1MaxPu  "Maximum reactive power in p.u (base SnRef) at terminal 1";
+    parameter Types.ReactivePowerPu Q2MinPu  "Minimum reactive power in p.u (base SnRef) at terminal 2";
+    parameter Types.ReactivePowerPu Q2MaxPu  "Maximum reactive power in p.u (base SnRef) at terminal 2";
+    parameter Real KLosses "Coefficient between 0 and 1 (no loss) modelling the losses in the HVDC";
+
+protected
+
+    parameter Types.ComplexVoltagePu u10Pu  "Start value of complex voltage at terminal 1 in p.u (base UNom)";
+    parameter Types.ComplexCurrentPu i10Pu  "Start value of complex current at terminal 1 in p.u (base UNom, SnRef) (receptor convention)";
+    parameter Types.ComplexApparentPowerPu s10Pu "Start value of complex apparent power at terminal 1 in p.u (base SnRef) (receptor convention)";
+
+    Types.ActivePowerPu P1Pu (start = s10Pu.re) "Active power at terminal 1 in p.u (base SnRef) (receptor convention)";
+    Types.ActivePowerPu P2Pu (start = 0) "Active power at terminal 2 in p.u (base SnRef) (receptor convention)";
+    Types.ActivePowerPu PInj1Pu (start = - s10Pu.re) "Active power at terminal 1 in p.u (base SnRef) (generator convention)";
+    Types.ActivePowerPu PInj2Pu (start = 0) "Active power at terminal 2 in p.u (base SnRef) (generator convention)";
+    Types.VoltageModulePu U1Pu (start = ComplexMath.'abs'(u10Pu)) "Voltage amplitude at terminal 1 in p.u (base UNom)";
+    Types.ComplexApparentPowerPu s1Pu(re (start = s10Pu.re), im (start = s10Pu.im)) "Complex apparent power at terminal 1 in p.u (base SnRef) (receptor convention)";
+    Types.ReactivePowerPu Q1Pu (start = s10Pu.im) "Reactive power at terminal 1 in p.u (base SnRef) (receptor convention)";
+    Types.ReactivePowerPu Q2Pu (start = 0) "Reactive power at terminal 2 in p.u (base SnRef) (receptor convention)";
+    Types.ReactivePowerPu QInj1Pu (start = - s10Pu.im) "Reactive power at terminal 1 in p.u (base SnRef) (generator convention)";
+    Types.ReactivePowerPu QInj2Pu (start = 0) "Reactive power at terminal 2 in p.u (base SnRef) (generator convention)";
+
+    Types.ReactivePowerPu Q1RawPu (start = s10Pu.im) "Raw reactive power at terminal 1 in p.u (base SnRef)";
+
+equation
+
+  Q1RawPu = tanPhi1Ref.value * P1Pu;
+
+  if Q1RawPu >= Q1MaxPu then
+   Q1Pu = Q1MaxPu;
+  elseif Q1RawPu <= Q1MinPu then
+   Q1Pu = Q1MinPu;
+  else
+   Q1Pu = Q1RawPu;
+  end if;
+
+   U1Pu = ComplexMath.'abs'(terminal1.V);
+   s1Pu = Complex(P1Pu, Q1Pu);
+   s1Pu = terminal1.V * ComplexMath.conj(terminal1.i);
+   P1Pu = P1RefPu;
+
+   P2Pu = 0;
+   Q2Pu = 0;
+   terminal2.i.re = 0;
+   terminal2.i.im = 0;
+
+   PInj1Pu = - P1Pu;
+   PInj2Pu = - P2Pu;
+   QInj1Pu = - Q1Pu;
+   QInj2Pu = - Q2Pu;
+
```

Comment:
I understand that you can't extend the base class model but perhaps, we can create a base dangling model that will be then used by both HVDC Dangling models, with only the reactive parts being different. What is your opinion? Any reason to discard this option?

In addition, it would be nice to add a few comments:
// Sign convention modification 
// Disconnected side 
etc.

Reply:
Ok, I'll create a base class for dangling models.
I'll also add some comments.